### PR TITLE
feat: add inert business calendars

### PR DESCRIPTION
**Pull request by JoelA510**
## Summary
- add non-default `weekday` and `us-federal-observed` app business calendars while keeping `defaultBusinessCalendar` on `calendar-day`
- mirror the inert calendars in Supabase Edge shared utilities
- add app, edge, and parity tests for weekday skipping, US federal observed holidays, negative offsets, zero offsets, weekend/year boundaries, and observed weekend holidays
- update date-engine ADR and architecture docs for the R4 slice

## Validation
- npx vitest --run Testing/unit/shared/lib/date-engine/business-calendar.test.ts Testing/unit/supabase/functions/_shared/business-calendar.test.ts Testing/unit/shared/lib/date-engine/parity.test.ts
- AGENT_MODE=true npm run verify-architecture
- npm run lint
- npm run typecheck:agent
- npx tsc --noEmit --pretty false
- npm test
- npm run build
- git diff --check

## Notes
- Runtime scheduling remains on `calendar-day`; PR R5 owns the behavior switch.

### CODE REVIEW

File: src/shared/lib/date-engine/business-calendar.ts [hunk 1]
```diff
@@ -89,6 +166,80 @@ const resolveBusinessDate = (input: BusinessCalendarDateInput | null | undefined
  return isValid(date) ? date : null;
 };
 
+const createSkippingBusinessCalendar = (
+ id: Exclude<BusinessCalendarId, 'calendar-day'>,
+ isExcludedDate: (date: Date) => boolean,
+): BusinessCalendar => {
```
[the remark this answers is not in the document]
Comment (JoelA510) [reply src/shared/lib/date-engine/business-calendar.ts:233]: Addressed in 1c90707e by documenting createSkippingBusinessCalendar with @param/@returns, along with the adjacent helper docs. I reran the targeted date-engine tests, lint, and typecheck after the doc update.
